js/app/modules/card/constrainedThumb.js

```diff
@@ -0,0 +1,255 @@
+            },
+            {
+                target_object: this._context_widget,
+                target_attribute: Emeus.ConstraintAttribute.BOTTOM,
+                source_attribute: Emeus.ConstraintAttribute.BOTTOM,
+            },
+            {
+                target_object: this._context_widget,
+                target_attribute: Emeus.ConstraintAttribute.WIDTH,
+                source_attribute: Emeus.ConstraintAttribute.WIDTH,
+            },
+            {
+                target_object: this._context_widget,
+                target_attribute: Emeus.ConstraintAttribute.LEFT,
+                source_attribute: Emeus.ConstraintAttribute.LEFT,
+            },
+        ];
+    },
+
+    _main_layout: function (image_portion_attr, image_full_attr) {
+        let constraints = [
+            {
+                target_object: this._thumbnail_frame,
+                target_attribute: Emeus.ConstraintAttribute.TOP,
+                source_attribute: Emeus.ConstraintAttribute.TOP,
+            },
+            {
+                target_object: this._thumbnail_frame,
+                target_attribute: Emeus.ConstraintAttribute.LEFT,
+                source_attribute: Emeus.ConstraintAttribute.LEFT,
+            },
+            {
+                target_object: this._thumbnail_frame,
+                target_attribute: image_portion_attr,
+                source_attribute: image_portion_attr,
+                multiplier: 0.66,
+            },
+            {
+                target_object: this._thumbnail_frame,
+                target_attribute: image_full_attr,
+                source_attribute: image_full_attr,
+            },
+            {
+                target_object: this._grid,
+                target_attribute: Emeus.ConstraintAttribute.RIGHT,
+                source_attribute: Emeus.ConstraintAttribute.RIGHT,
+            },
+            {
+                target_object: this._grid,
+                target_attribute: Emeus.ConstraintAttribute.BOTTOM,
+                source_attribute: Emeus.ConstraintAttribute.BOTTOM,
+            },
+            {
+                target_object: this._grid,
+                target_attribute: image_portion_attr,
+                source_attribute: image_portion_attr,
+                multiplier: 0.34,
+            },
+            {
+                target_object: this._grid,
+                target_attribute: image_full_attr,
+                source_attribute: image_full_attr,
+            },
+        ];
+        constraints.forEach((props) => this._add_constraint(props, this._layout));
+    },
+
+    vfunc_size_allocate: function (alloc) {
+        this._layout.clear_constraints();
+        this._text_layout.clear_constraints();
+        let orientation = this._get_orientation(alloc.width, alloc.height);
+
+        let show_synopsis = this._should_show_synopsis(alloc.width, alloc.height, orientation);
+
+        let show_context = !(this._should_hide_context(alloc.width, alloc.height));
```

Comment:
nit: for consistency it would be better to name _should_hide_context()  _should_show_context()